doc/OnlineDocs/explanation/analysis/alternative_solutions.rst

```diff
@@ -2,233 +2,359 @@
 Generating Alternative (Near-)Optimal Solutions
 ###############################################
 
+.. py:currentmodule:: pyomo.contrib.alternative_solutions
+
 Optimization solvers are generally designed to return a feasible solution
 to the user. However, there are many applications where a user needs
 more context than this result. For example,
 
-* alternative solutions can support an assessment of trade-offs between
-  competing objectives;
-
-* if the optimization formulation may be inaccurate or untrustworthy,
-  then comparisons amongst alternative solutions provide additional
-  insights into the reliability of these model predictions; or
-
-* the user may have unexpressed objectives or constraints, which only
-  are realized in later stages of model analysis.
+* alternative optimal solutions can be used to assess trade-offs between
+  competing objectives; 
+ 
+* comparisons amongst alternative solutions provide 
+  insights into the efficacy of model predictions with 
+  inaccurate or untrusted optimization formulations; or 
+ 
+* alternative solutions can be identified to support the future analysis of model revisions (e.g. to 
+  account for previously unexpressed constraints).
 
 The *alternative-solutions library* provides a variety of functions that
 can be used to generate optimal or near-optimal solutions for a pyomo
 model. Conceptually, these functions are like pyomo solvers. They can
-be configured with solver names and options, and they return a list of
+be configured with solver names and options, and they return a pool of
 solutions for the pyomo model. However, these functions are independent
-of pyomo's solver interface because they return a custom solution object.
+of pyomo's solver interface because they return a custom pool manager object.
 
 The following functions are defined in the alternative-solutions library:
 
-* ``enumerate_binary_solutions``
+* :py:func:`enumerate_binary_solutions`
 
     * Finds alternative optimal solutions for a binary problem using no-good cuts.
 
-* ``enumerate_linear_solutions``
+* :py:func:`enumerate_linear_solutions`
 
     * Finds alternative optimal solutions for a (mixed-integer) linear program.
```

Comment:
This is the only one that doesn't say *how* it generates alternative optimal solutions.